app/models/taxon_lineage.rb

```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```

Comment:
Maybe add a comment for this line...
What is the impact of this?

Reply:
This magic comment needs to be at the very first line of the source file, and it is used to implicitly mark all string literals as frozen.
It provides better resource usage, since Ruby knows that is safe to reuse string literals across multiple calls because it is guaranteed the string cannot be changed.
The original proposal can be found here: https://bugs.ruby-lang.org/issues/8976#note-30

Reply:
(by the way, the original PR description mentions it:

> 5) Added # frozen_string_literal: true to this module to save resources